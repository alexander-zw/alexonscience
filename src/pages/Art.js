import React from 'react';
import '../styles/index.css';
import '../styles/Art.css';
import hello_there from '../images/hello_there.png';
import general_kenobi from '../images/general_kenobi.png';
import math_name from '../images/math_name.jpg';
import fancy_pants_sketch from '../images/fancy_pants_sketch.jpg';

const Art = () => {
    return (
        <div>
            <div className="text-div">
                <p>
                    Here is some random artwork I have drawn for fun. Enjoy!
                </p>
            </div>

            <div className="artwork-div">
                <div id="kenobi-image">
                    <img className="bottom artwork-image" src={general_kenobi} alt="Hello there"/>
                    <img className="top artwork-image" src={hello_there} alt="General Kenobi"/>
                </div>
            </div>

            <div className="artwork-div">
                <img className="artwork-image" src={math_name} alt="How to name a math concept"/>
            </div>

            <div className="artwork-div" id="fancy-pants-image">
                <img className="artwork-image" id="fancy-pants-image"
                src={fancy_pants_sketch} alt="Fancy pants sketch"/>
            </div>
        </div>
    );
}

export default Art;
