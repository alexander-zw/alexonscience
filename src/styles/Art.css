#kenobi-image img {
    position: absolute;
    width: 50%;
    left: 50%;
    transform: translate(-50%,0);
    background-color: #f8f8f8;
    -webkit-transition: opacity 1s ease-in-out;
    -moz-transition: opacity 1s ease-in-out;
    -o-transition: opacity 1s ease-in-out;
    transition: opacity 1s ease-in-out;
}

#kenobi-image img.top:hover {
    opacity:0;
}
