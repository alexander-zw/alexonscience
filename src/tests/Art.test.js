/**
 * Test for the Art.js component.
 */
import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Art from "../components/Art";

describe("Art", () => {
    test("renders correctly", () => {
        const tree = renderer.create(<Art />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
