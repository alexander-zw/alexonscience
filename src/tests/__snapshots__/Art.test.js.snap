// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Art renders correctly 1`] = `
<div>
  <div
    className="text-div"
  >
    <p>
      Here is some random artwork I have drawn for fun. Enjoy!
    </p>
  </div>
  <div
    className="artwork-div"
  >
    <div
      id="kenobi-image"
    >
      <img
        alt="Hello there"
        className="bottom artwork-image"
        src="general_kenobi.png"
      />
      <img
        alt="General Kenobi"
        className="top artwork-image"
        src="hello_there.png"
      />
    </div>
  </div>
  <div
    className="artwork-div"
  >
    <img
      alt="How to name a math concept"
      className="artwork-image"
      src="math_name.jpg"
    />
  </div>
  <div
    className="artwork-div"
    id="fancy-pants-image"
  >
    <img
      alt="Fancy pants sketch"
      className="artwork-image"
      id="fancy-pants-image"
      src="fancy_pants_sketch.jpg"
    />
  </div>
</div>
`;
